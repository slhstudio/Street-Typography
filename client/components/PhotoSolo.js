import React from 'react';
import Edit from './Edit';

const PhotoSolo = (props) => {
  return (
    <div className='photoSoloBox'>  
      <img className= 'photoSolo' src={`/uploads/${props.photo}`}/>
      <img className= 'map' src={props.map}/>
        { props.name === props.author
          ? <div>
              <Edit 
                info={props.notes}
                photo={props.photo}
              />
            </div>
          : <p>{props.notes}</p>
        }
    </div>
  )
}

export default PhotoSolo;


