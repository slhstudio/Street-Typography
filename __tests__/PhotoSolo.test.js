import React from 'react';
import PhotoSolo from '../client/components/PhotoSolo';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';


const name1 = 'me';
const name2 = 'notMe';
const author = 'me';
const notes = 'xyz';
const photo = 'photo1';
const map = 'Paris, Texas'

describe('<PhotoSolo/>', () => {
  it ('renders and displays properly', () => {
     const wrapper = shallow(<PhotoSolo photo={photo} map={map}/>);
     expect(toJSON(wrapper)).toMatchSnapshot();
    //  expect(wrapper.find('.photoSoloBox').exists()).toBe(true);
    //  expect(wrapper.find('img').first().props().src).toBe(`/uploads/photo1`);
    //  expect(wrapper.find('img').last().props().src).toBe('Paris, Texas');
  });
  it ('if author is same as user, edit form renders', () => {
    const wrapper = shallow(<PhotoSolo name={name1} author={author} notes={notes}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    //expect(wrapper.find('Edit').exists()).toBe(true);
  });
  it ('else only the notes render', () => {
    const wrapper = shallow(<PhotoSolo name={name2} author={author} notes={notes}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    // expect(wrapper.find('p').exists()).toBe(true);
    // expect(wrapper.find('p').text()).toBe('xyz');
  });
});