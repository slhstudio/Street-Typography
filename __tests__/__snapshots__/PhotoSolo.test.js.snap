// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoSolo/> else only the notes render 1`] = `
<div
  className="photoSoloBox"
>
  <img
    className="photoSolo"
    src="/uploads/undefined"
  />
  <img
    className="map"
  />
  <p>
    xyz
  </p>
</div>
`;

exports[`<PhotoSolo/> if author is same as user, edit form renders 1`] = `
<div
  className="photoSoloBox"
>
  <img
    className="photoSolo"
    src="/uploads/undefined"
  />
  <img
    className="map"
  />
  <div>
    <Edit
      info="xyz"
    />
  </div>
</div>
`;

exports[`<PhotoSolo/> renders and displays properly 1`] = `
<div
  className="photoSoloBox"
>
  <img
    className="photoSolo"
    src="/uploads/photo1"
  />
  <img
    className="map"
    src="Paris, Texas"
  />
  <div>
    <Edit
      photo="photo1"
    />
  </div>
</div>
`;
